### DISCUSSION TITLE
GitHub Backlinks/Pings due to embedding release notes in commit messages

### DISCUSSION THREAD

**Discussion by Zamiell** [renovatebot/renovate#31230] [Request Help]
Hello; I am an open-source contributor and I got a spurious ping from the Renovate bot:

https://github.com/mmkal/eslint-plugin-mmkal/commit/db3278d1962372f00b34dd51ce8fe1685ee67ee6

**Comment by rarkins** [comment]
This is caused by a custom configuration by @mmkal and is not _default_ Renovate behavior. On the other hand this is not _wrong_ to embed release notes in commit messages.

Frankly I think this is undesirable behavior by GitHub itself (where @ mentions in commit messages on public repos leads to pings).

We attempted to fix this previously: https://github.com/renovatebot/renovate/pull/30842 but we needed to revert the change: https://github.com/renovatebot/renovate/pull/30910 due to https://github.com/renovatebot/renovate/pull/30842#issuecomment-2298437335

I feel like we're in a tough situation where:
- We can't simply escape all `@` in release notes because it can break other things, but
- We would be playing a game of whack-a-mole trying to find safe patterns to escape, starting with `[@<github username regex>]`

**Reply by mmkal** [reply]
Whoops, sorry about that. In my particular case I would be fine to escape all `@`s if it would help - maybe renovate could add an opt-in  escaping function I could add to the template?

**Reply by rarkins** [reply]
Perhaps as a handlebars plugin? _maybe_ the existing replace function can already do it, if we simply replace `"@"` with `"@ "`

**Reply by mmkal** [reply]
@rarkins what about `@` -> `` `@` ``? I think that would mostly preserve copy-pastability too.

- Github usernames: `@`mmkal
- Emails: hello`@`example.com

If that would work for you, I could open a PR if you could point me in roughly the right direction? (I can probably figure out how to add if you can just let me know a general area of code to start looking in)

**Reply by viceice** [reply]
I see wrong encoded stuff too , I think we need to parse the markdown to better detect real users and only encode those.

you can see wrong encoded in code block in PR below 

https://github.com/containerbase/base/pull/3279

**Reply by rarkins** [reply]
@viceice isn't that the opposite problem though? OP thinks that renovate doesn't escape enough, you point out a case where it escapes too much

**Reply by viceice** [reply]
yes, I just searched a related topic. It should show that the escaping isn't as trivial as it looks.

**Comment by Zamiell** [comment]
I am getting yet more pings today:
- https://github.com/mmkal/eslint-plugin-mmkal/commit/d1b5c12ef3fab3f1987b3cec321d30ea4dfe65c5
- https://github.com/mmkal/eslint-plugin-mmkal/commit/28303c8c84fa6133b506223a0f8c709204712d0e
- https://github.com/mmkal/eslint-plugin-mmkal/commit/cc4ec7caa8bd12cd570f0291a50e902ffbc2349f
- https://github.com/mmkal/eslint-plugin-mmkal/commit/0d47e995c35a0800c66996ce82fa3fd961375987

@mmkal Please turn this off.
@rarkins I think that the renovatebot code should have a check to make it impossible to misconfigure it in this way.

**Reply by mmkal** [reply]
Ok, I'll turn this off.

**Reply by mmkal** [reply]
Done. Please let me know if you get any more.

I agree that this shouldn't happen so easily - although I think it's worth noting, it's not even misconfigured, as @rarkins says it's just a bad GitHub feature that `@whatever` will ping someone. @rarkins I'll reply on the other thread about how to escape - I am disappointed to have turned this off because now my dependency updates have too little info about what's contained in them. The release notes were very useful to me, despite the pings.

**Reply by rarkins** [reply]
@mmkal please start a new discussion. Maybe we can work out a sanitation template plugin to capture the most common backlink causes
